crates/pbs/src/mev_boost/register_validator.rs

```diff
@@ -63,15 +64,61 @@ pub async fn register_validator<S: BuilderApiState>(
     }
 }
 
-#[tracing::instrument(skip_all, name = "handler", fields(relay_id = relay.id.as_ref()))]
-async fn send_register_validator(
+/// Register validator to relay, retry connection errors until the
+/// given timeout has passed
+async fn send_register_validator_with_timeout(
     registrations: Vec<ValidatorRegistration>,
     relay: RelayClient,
     headers: HeaderMap,
     timeout_ms: u64,
 ) -> Result<(), PbsError> {
     let url = relay.register_validator_url()?;
+    let mut remaining_timeout_ms = timeout_ms;
+    let mut retry = 0;
+    let mut backoff = Duration::from_millis(250);
+
+    loop {
```

Comment:
can we use the reqwest middleware?

https://docs.rs/reqwest-retry/latest/reqwest_retry/